tests/std/tests/P0896R4_common_iterator_death/test.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CONTAINER_DEBUG_LEVEL 1
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <iterator>
+
+#include <test_death.hpp>
+using namespace std;
+
+struct simple_input_iter {
+    using value_type      = int;
+    using difference_type = int;
+
+    value_type operator*() const {
+        return 0;
+    }
+    value_type operator->() const {
+        return 0;
+    }
+    simple_input_iter& operator++() {
+        return *this;
+    }
+    simple_input_iter operator++(int) {
+        return *this;
+    }
+
+    bool operator==(simple_input_iter const&) const = default;
+    bool operator==(const default_sentinel_t&) const {
+        return true;
+    }
+
+    friend void iter_swap(const simple_input_iter&, const simple_input_iter&) {}
+};
+
+using CIT = common_iterator<simple_input_iter, default_sentinel_t>;
+
+void test_case_operator_dereference() {
+    CIT cit{default_sentinel};
+    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_dereference_const() {
+    const CIT cit{default_sentinel};
+    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_arrow() {
+    CIT cit{default_sentinel};
+    (void) (cit.operator->()); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_preincrement() {
+    CIT cit{default_sentinel};
+    ++cit; // cannot pre increment common_iterator that holds a sentinel
+}
+
+void test_case_operator_increment() {

```

Comment:
Should this be `postincrement` for consistency?

Reply:
Yes it should